debugger/checkpoints.ml

```diff
@@ -75,12 +77,13 @@ let current_time () =
 let current_report () =
   !current_checkpoint.c_report
 
-let current_pc () =
-  match current_report () with
-    None | Some {rep_type = Exited | Uncaught_exc} -> None
-  | Some {rep_program_pointer = pc } -> Some pc
-
 let current_pc_sp () =
   match current_report () with
-    None | Some {rep_type = Exited | Uncaught_exc} -> None
-  | Some {rep_program_pointer = pc; rep_stack_pointer = sp } -> Some (pc, sp)
+  | Some {rep_type = Event | Breakpoint;
+          rep_program_pointer = pc; rep_stack_pointer = sp } -> Some (pc, sp)
+  | _ -> None
```

Comment:
I think the way the current piece of code is written is right:
- the pattern-matching corresponds to the cases where PC/SP is given in debugger.c (see lines 419-430 of debugger.c)
- returning [None] is the "safe" option (no pc nor stack frame will be printed by the debugger, instead of printing garbage/segfault)

If you prefer, I can make it more robust by making it exhaustive, but this is asymmetric with what happens in debugger.c.

Reply:
That makes sense, thanks! Maybe a comment to point at that logic (I guess not line numbers for robustness; "the `if (event == ...)` conditional before the `command_loop:` label in `caml_debugger` in debugger.c" may do.)